File: CellRemoval/solve/cpp/CellRemovalSolver.cpp
#include <fstream>
#include <topcoder>
#include "CellRemoval.cpp"
namespace tc = TopCoder;

int main(int argc, char const *argv[]) {
  try {
    std::ifstream ifs(argv[1]);
    vector<int> parent; tc::read(ifs, parent); tc::next(ifs);
    int deletedCell; tc::read(ifs, deletedCell);
    ifs.close();

    std::ofstream ofs(argv[2]);
    CellRemoval solver;
    tc::show(ofs, solver.cellsLeft(parent, deletedCell));
    ofs.close();
  } catch (std::exception &e) {
    std::cerr << e.what() << std::endl;
  }
  return 0;
}

File: CellRemoval/solve/java/CellRemovalSolver.java
import java.io.FileReader;
import java.io.FileWriter;

import java.util.List;
import java.util.ArrayList;

import org.topcoder.TopcoderReader;
import org.topcoder.TopcoderWriter;
import org.topcoder.TypeRef;

public class CellRemovalSolver {
    public static void main(String[] args) {
    try {
        TopcoderReader reader = new TopcoderReader(new FileReader(args[0]));
        List<Integer> parentBoxed = (List<Integer>) reader.next(new TypeRef<List<Integer>>(){}.getType());
        int[] parent = new int[parentBoxed.size()];
        for (int _i = 0; _i < parentBoxed.size(); ++_i)
            parent[_i] = parentBoxed.get(_i);
        reader.next();
        
        int deletedCell = (Integer) reader.next(Integer.class);
        reader.close();

        CellRemoval solver = new CellRemoval();
        TopcoderWriter writer = new TopcoderWriter(new FileWriter(args[1]));
        writer.write(solver.cellsLeft(parent, deletedCell));
        writer.close();
    } catch (Exception err) {
        err.printStackTrace(System.err);
    }
    }
}
